### PR TITLE
Upgrade core

### PR DESCRIPTION
**Pull request by swansontec**
### CHANGELOG

Does this branch warrant an entry to the CHANGELOG?

- [x] Yes
- [ ] No

### Dependencies

 none

### Description

 none

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1207134573674963

### CODE REVIEW

File: src/common/utxobased/info/bitcoin.ts [hunk 1]
```diff
@@ -23,10 +22,6 @@ const currencyInfo: EdgeCurrencyInfo = {
   addressExplorer: 'https://blockchair.com/bitcoin/address/%s',
   transactionExplorer: 'https://blockchair.com/bitcoin/transaction/%s',
 
-  // Images:
-  symbolImage: `${IMAGE_SERVER_URL}/bitcoin-logo-solo-64.png`,
-  symbolImageDarkMono: `${IMAGE_SERVER_URL}/bitcoin-logo-solo-64.png`,
```
Comment (samholmes) [inline src/common/utxobased/info/bitcoin.ts:28]: Ah nice, I just checked that this isn't used in the GUI either.

File: src/common/utxobased/db/Processor.ts [hunk 2]
```diff
@@ -368,28 +368,12 @@ export async function makeProcessor(
...
-              '',
-              unixTime(startDate.getTime()),
-              unixTime(endDate.getTime())
-            )
-          }
+          const txData: TxIdByDate[] = await tables.txIdsByDate.query(
+            '',
+            0,
+            unixTime(Date.now())
+          )
```
Comment (samholmes) [inline src/common/utxobased/db/Processor.ts:376]: How is this change relevant to the upgrade I wonder?

File: test/common/utxobased/db/Processor.spec.ts [hunk 3]
```diff
@@ -589,21 +589,10 @@ describe('Processor transactions tests', () => {
...
-    })
-    expect(results).to.deep.equal([])
-
-    const [tx6] = await processor.fetchTransactions({
-      options: {
-        startDate: new Date(9_000),
-        endDate: new Date(15_000)
-      }
+    const [tx4] = await processor.fetchTransactions({
+      options: { tokenId: null }
```
Comment (samholmes) [inline test/common/utxobased/db/Processor.spec.ts:593]: Are we deprecating some internal API without specifying the deprecation?

File: test/common/utxobased/engine/engine.spec.ts [hunk 4]
```diff
@@ -286,30 +286,21 @@ describe('engine.spec', function () {
...
       })
-
-      it('Should get transactions from cache with options', async function () {
-        const options: EdgeGetTransactionsOptions = {
-          startIndex: 1,
-          startEntries: 2
-        }
-        const txs = await engine.getTransactions(options)
-        assert.equal(txs.length, 2, 'should have 2 tx from cache')
-      })
```
Comment (samholmes) [inline test/common/utxobased/engine/engine.spec.ts:310]: Why was this test removed?

**Review by samholmes — CHANGES_REQUESTED** [review]
More questions again

File: test/common/utxobased/db/Processor.spec.ts (hunk 3 above)
Comment (swansontec) [reply test/common/utxobased/db/Processor.spec.ts:593]: The core removed these features, and the removals were all documented in the release notes. Specifically, we removed all the options to `getTransactions` besides the `tokenId`, so the core is in charge of handling the pagination, searching, and other advanced query options.

File: test/common/utxobased/engine/engine.spec.ts (hunk 4 above)
Comment (swansontec) [reply test/common/utxobased/engine/engine.spec.ts:310]: See above. The core removed the ability to pass any `getTransactions` flags besides the `tokenId`. Since we don't need these features, we don't need to test them.

File: src/common/utxobased/engine/types.ts [hunk 5]
```diff
@@ -42,13 +42,15 @@ export const asUtxoSignMessageOtherParams = asObject({
...
+const asOutputSort = asValue('bip69', 'targets')
+
 export type UtxoSpendInfoOtherParams = ReturnType<
   typeof asUtxoSpendInfoOtherParams
 >
 export const asUtxoSpendInfoOtherParams = asObject({
   enableRbf: asOptional(asBoolean),
   forceChangeAddress: asOptional(asString),
-  outputSort: asOptional(asValue('bip69', 'targets'), 'bip69'),
+  outputSort: asOptional(asOutputSort, 'bip69'),
```
[the remark this answers is not in the document]
Comment (swansontec) [reply src/common/utxobased/engine/types.ts:53]: Added the message: "When the asValue cleaner gets embedded inside asObject, it will often infer string instead of the specific literal type. Moving the asValue cleaner to its own line allows TypeScript to infer the exact string literals, instead of the string catch-all."

File: test/common/utxobased/db/Processor.spec.ts (hunk 3 above)
Comment (samholmes) [reply test/common/utxobased/db/Processor.spec.ts:593]: Ah, cause the processor coupled its type with the core originally, the core upgrade impacts the processor API. Got it.

File: src/common/utxobased/db/Processor.ts (hunk 2 above)
Comment (samholmes) [reply src/common/utxobased/db/Processor.ts:376]: My final question now is: is there a new protocol for the core to query the plugin for paging or streaming of this data that we should mark as a TODO?

**Review by samholmes — APPROVED** [review]
Approved, but with one final question or cosideration.
